src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/RestoreOperation/RestoreDatabaseTaskDataObject.cs

```diff
@@ -921,8 +921,9 @@ public void CreateNewRestorePlan()
 
         public bool ShouldCreateNewPlan()
         {
-            return string.Compare(RestorePlanner.DatabaseName, this.RestoreParams.GetOptionValue<string>(RestoreOptionsHelper.SourceDatabaseName), StringComparison.InvariantCultureIgnoreCase) != 0 ||
+            return RestorePlan == null || string.Compare(RestorePlanner.DatabaseName, this.RestoreParams.GetOptionValue<string>(RestoreOptionsHelper.SourceDatabaseName), StringComparison.InvariantCultureIgnoreCase) != 0 ||
```

Comment:
>GetOptionValue [](start = 105, length = 14)

it would be nice if RestoreParams had wrapper methods to get at common option values to make the code a little more readable.  It looks like there are some already, like for BackupFilePaths below.

Reply:
will make it more readable in the next PR 

---
In reply to: [141914116](https://github.com/Microsoft/sqltoolsservice/pull/470#discussion_r141914116) [](ancestors = 141914116)